src/fastga/models/aerodynamics/external/vlm/vlm.py

```diff
@@ -1140,3 +1118,420 @@ def read_results(result_file_path):
+        panel_span[n_y:] = panel_span[:n_y]
+
+        # Calculate characteristic points (Right and left side)
+        for i in range(n_x):
+            panel_chord[i * n_y : (i + 1) * n_y] = 0.5 * (
+                (x_panel[i + 1, :n_y] - x_panel[i, :n_y])
+                + (x_panel[i + 1, 1 : n_y + 1] - x_panel[i, 1 : n_y + 1])
+            )
+            panel_surf[i * n_y : (i + 1) * n_y] = (
+                panel_span[:n_y] * panel_chord[i * n_y : (i + 1) * n_y]
+            )
+            # right side
+
+            # point A for the right-half
+            x_1[i * n_y : (i + 1) * n_y] = x_panel[i, :n_y] + 0.25 * (
+                x_panel[i + 1, :n_y] - x_panel[i, :n_y]
+            )
+
+            y_1[i * n_y : (i + 1) * n_y] = y_panel[:n_y]
+
+            # point B for the right-half
+            x_2[i * n_y : (i + 1) * n_y] = x_panel[i, 1 : n_y + 1] + 0.25 * (
+                x_panel[i + 1, 1 : n_y + 1] - x_panel[i, 1 : n_y + 1]
+            )
+
+            y_2[i * n_y : (i + 1) * n_y] = y_panel[1 : n_y + 1]
+
+            # point P for the right-half
+            x_c[i * n_y : (i + 1) * n_y] = (
+                x_panel[i, :n_y] + x_panel[i, 1 : n_y + 1]
+            ) * 0.5 + 0.75 * panel_chord[i * n_y : (i + 1) * n_y]
+
+            y_c[i * n_y : (i + 1) * n_y] = (y_panel[:n_y] + y_panel[1 : n_y + 1]) * 0.5
+
+            # left side
+
+            # point A for the left-half
+            x_1[n_x * n_y + i * n_y : n_x * n_y + (i + 1) * n_y] = x_panel[
+                i, n_y + 1 : 2 * n_y + 1
+            ] + 0.25 * (x_panel[i + 1, n_y + 1 : 2 * n_y + 1] - x_panel[i, n_y + 1 : 2 * n_y + 1])
+
+            y_1[n_x * n_y + i * n_y : n_x * n_y + (i + 1) * n_y] = y_panel[n_y + 1 : 2 * n_y + 1]
+
+            # point B for the left-half
+            x_2[n_x * n_y + i * n_y : n_x * n_y + (i + 1) * n_y] = np.concatenate(
+                (
+                    np.array([x_panel[i, 0] + 0.25 * (x_panel[i + 1, 0] - x_panel[i, 0])]),
+                    (
+                        x_panel[i, n_y : 2 * n_y]
+                        + 0.25 * (x_panel[i + 1, n_y : 2 * n_y] - x_panel[i, n_y : 2 * n_y])
+                    )[1:],
+                )
+            )
+            y_2[n_x * n_y + i * n_y : n_x * n_y + (i + 1) * n_y] = np.concatenate(
+                (
+                    np.array([0.0]),
+                    y_panel[n_y + 1 : 2 * n_y],
+                )
+            )
+        # Calculate remaining characteristic points (Left side)
+        # point P for the left-half
+        x_c[n_x * n_y :] = x_c[: n_x * n_y]
+        y_c[n_x * n_y :] = -y_c[: n_x * n_y]
+
+        return (
+            x_panel,
+            y_panel,
+            x_le,
+            chord,
+            panel_chord,
+            panel_span,
+            panel_surf,
+            x_1,
+            y_1,
+            x_2,
+            y_2,
+            x_c,
+            y_c,
+        )
+
+
+def string_to_array(arr):
```

Comment:
This function is redefined in openvsp and in xfoil. Could be avoided?

Reply:
Corrected in c178125
Also let me explain the reason for this function and I'll take the example of Xfoil for that. For Xfoil, the reasults we store are  under the form of an array of alpha and an array of Cl/Cd, ... So for each run of Xfoil (one column of a dataframe) we have one cell that must contain an array. Except that when storing an simple array in a dataframe, it will create a column for each element of the array. The way we chose to solve that is to convert that array as a string before storing and back to an array when looking up the results.